File: python/ops-hub-heartbeat.py
#!/usr/bin/env python3
# ops-hub-heartbeat.js — portiert nach python
# Quelle: javascript, OpenClaw@gateway1:scripts/ops-hub-heartbeat.js
# auch in: OpenClaw@gateway2:scripts/ops-hub-heartbeat.js
# Erzeugt: 2026-08-09 durch ABSTRACTIONS_MANAGER.py

import os
import datetime
import sys

# Aktualisiere den Statusbericht mit aktueller Zeit
status_path = os.path.join(os.path.dirname(__file__), '../docs/ops-hub/status.md')

def update_heartbeat():
    try:
        with open(status_path, 'r', encoding='utf-8') as file:
            content = file.read()
    except Exception as err:
        print(f'❌ Konnte status.md nicht lesen: {err}', file=sys.stderr)
        return

    now = datetime.datetime.now(datetime.timezone(datetime.timedelta(hours=1))).strftime('%d.%m.%Y, %H:%M:%S')
    updated = content.replace('(Letzter Heartbeat:) [^\n]*', f'Letzter Heartbeat: {now}', 1)

    try:
        with open(status_path, 'w', encoding='utf-8') as file:
            file.write(updated)
        print(f'✅ Heartbeat aktualisiert: {now}')
    except Exception as err:
        print(f'❌ Konnte status.md nicht schreiben: {err}', file=sys.stderr)

update_heartbeat()

File: perl5/ops-hub-heartbeat.pl
#!/usr/bin/perl
# ops-hub-heartbeat.js — portiert nach perl5
# Quelle: javascript, OpenClaw@gateway1:scripts/ops-hub-heartbeat.js
# auch in: OpenClaw@gateway2:scripts/ops-hub-heartbeat.js
# Erzeugt: 2026-08-09 durch ABSTRACTIONS_MANAGER.py

use strict;
use warnings;
use utf8;
use File::Spec;
use POSIX qw(strftime);

# Aktualisiere den Statusbericht mit aktueller Zeit
my $statusPath = File::Spec->catfile(File::Spec->updir(), 'docs', 'ops-hub', 'status.md');

sub updateHeartbeat {
    my $content;
    open(my $fh, '<:encoding(UTF-8)', $statusPath) or do {
        warn "❌ Konnte status.md nicht lesen: $!";
        return;
    };
    {
        local $/;
        $content = <$fh>;
    }
    close($fh);

    my $now = strftime("%d.%m.%Y, %H:%M:%S", localtime());
    $content =~ s/(Letzter Heartbeat:) [^\n]*/$1 $now/;

    open($fh, '>:encoding(UTF-8)', $statusPath) or do {
        warn "❌ Konnte status.md nicht schreiben: $!";
        return;
    };
    print $fh $content;
    close($fh);
    
    print "✅ Heartbeat aktualisiert: $now\n";
}

updateHeartbeat();
